plugins/modules/docker_secret.py

```diff
@@ -232,6 +279,10 @@ def create_secret(self):
         labels = {
             'ansible_key': self.data_key
         }
+        if self.rolling_versions:
+            self.version += 1
+            labels['ansible_version'] = str(self.version)
+            self.name = '{name}_v{version}'.format(name=self.name, version=self.version)
```

Comment:
For the above to work correctly, you need to add something here though.
```suggestion
            self.name = '{name}_v{version}'.format(name=self.name, version=self.version)
            self.secrets.append(...)
```

Reply:
Yeah, I was thinking about doing this, but to get the correct structure we'd have to run a `client.get_secret()` which seemed to me to be more complex than what's necessary.